include/exadg/postprocessor/time_control.cpp

```diff
@@ -0,0 +1,162 @@
+{
+  bool const interval = (data.trigger_interval > 0.0);
+  bool const timestep = Utilities::is_valid_timestep(data.trigger_every_time_steps);
+
+  // only one of both can be true
+  if(interval && timestep)
+    return TimeControlData::EvalType::Invalid;
+
+  if(interval)
+    return TimeControlData::EvalType::Interval;
+
+  if(timestep)
+    return TimeControlData::EvalType::Timestep;
+
+  return TimeControlData::EvalType::Invalid;
+}
+
+TimeControlData::TimeControlData()
+  : is_active(false),
+    start_time(std::numeric_limits<double>::max()),
+    end_time(std::numeric_limits<double>::max()),
+    trigger_interval(-1.0),
+    trigger_every_time_steps(numbers::invalid_timestep),
+    counter_start(0)
+{
+}
+
+void
+TimeControlData::print(dealii::ConditionalOStream & pcout, bool const unsteady) const
+{
+  if(unsteady || is_active)
+  {
+    print_parameter(pcout, "TimeControl start time", start_time);
+    print_parameter(pcout, "TimeControl end_time", end_time);
+    if(counter_start != 0)
+      print_parameter(pcout, "Counter start", counter_start);
+    if(get_unsteady_evaluation_type(*this) == TimeControlData::EvalType::Interval)
+      print_parameter(pcout, "TimeControl triggers every interval", trigger_interval);
+    if(get_unsteady_evaluation_type(*this) == TimeControlData::EvalType::Timestep)
+      print_parameter(pcout, "TimeControl triggers every timestep", trigger_every_time_steps);
+  }
+}
+
+TimeControl::TimeControl() : EPSILON(1.0e-10), reset_counter(true), counter(0),
+{
+}
+
+void
+TimeControl::setup(TimeControlData const & time_control_data_in)
+{
+  time_control_data = time_control_data_in;
+  counter           = time_control_data_in.counter_start;
+}
+
+// If setup() is not called needs_evaluation() will always evaluate to false
+bool
+TimeControl::needs_evaluation(double const time, types::time_step const time_step_number) const
+{
+  if(time_control_data.is_active == false)
+    return false;
+
+  // steady case
+  if(!Utilities::is_unsteady_timestep(time_step_number))
+  {
+    ++counter;
+    return true;
+  }
+
+  // unsteady evaluation
+  AssertThrow(get_unsteady_evaluation_type(time_control_data) != TimeControlData::EvalType::Invalid,
+              dealii::ExcMessage(
+                "Unsteady evaluation not possible since TimeControlData::EvalType::Invalid"));
+
+  if(time < time_control_data.start_time - EPSILON)
+    return false;
+
+  if(time > time_control_data.end_time + EPSILON)
+    return false;
+
+  bool evaluate = false;
+  if(get_unsteady_evaluation_type(time_control_data) == TimeControlData::EvalType::Timestep)
+  {
+    if((time_step_number - 1) % time_control_data.trigger_every_time_steps == 0)
+    {
+      evaluate = true;
+      ++counter;
+    }
+  }
+  else if(get_unsteady_evaluation_type(time_control_data) == TimeControlData::EvalType::Interval)
+  {
+    // In case of a restart, time might be significantly larger than start_time in the first time
```

Comment:
this increases the counter in case of a restart, starting e.g. with `counter = 13` (due to a previous simulation from which we want to restart), which is then incremented by one again and again during the simulation.
Hence, can we remove counter_start?

Reply:
Absolutely, I thought the user has to do it manually